src/bus/event_bus.cpp

```diff
@@ -0,0 +1,9 @@
+#include "bus/event_bus.hpp"
+
+namespace mesos {
+namespace serenity {
+
+EventBus* EventBus::instance = nullptr;
```

Comment:
I think we should have CommunicationBus class, and SerenityBus singleton to split implementation from usage.

But we need singleton to subscribe/publish to same bus in different modules (qos/re), right?